### PR TITLE
Result service - csv export of results

### PR DESCRIPTION
**Pull request by arnost00**
Added CSV result export in result service - single file CSV or one file per class

### CODE REVIEW

File: libquickevent/libquickeventcore/src/exporters/stageresultscsvexporter.cpp [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+	qfDebug() << "loading clases:" << qs;
+
+	QString sub_dir = QString("E%1").arg(currentStage());
+	QDir csv_dir(outDir());
+	csv_dir.mkpath(sub_dir);
+
+	if (single_file) {
+		QFile fCSV(csv_dir.absolutePath() + '/' + sub_dir + '/' + "results.csv");
+		qfInfo() << "Generating:" << fCSV.fileName();
+		if (!fCSV.open(QFile::WriteOnly | QFile::Text))
```
Comment (fvacek) [inline libquickevent/libquickeventcore/src/exporters/stageresultscsvexporter.cpp:55]: Please do not use TEXT open mode, we will get different files on different OS this way. https://doc.qt.io/qt-6/qiodevicebase.html#OpenModeFlag-enum

**Review by fvacek — CHANGES_REQUESTED** [review]

File: libquickevent/libquickeventcore/src/exporters/stageresultscsvexporter.h [hunk 2]
```diff
@@ -0,0 +1,28 @@
+#pragma once
+
+#include "fileexporter.h"
+#include <QTextStream>
```
[the remark this answers is not in the document]
Comment (arnost00) [reply libquickevent/libquickeventcore/src/exporters/stageresultscsvexporter.h:4]: Ok

File: libquickevent/libquickeventcore/src/exporters/stageresultscsvexporter.cpp [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+		qs += " WHERE " + where;
+	qs += " ORDER BY name";
+	qfDebug() << "loading clases:" << qs;
+
+	QString sub_dir = QString("E%1").arg(currentStage());
+	QDir csv_dir(outDir());
+	csv_dir.mkpath(sub_dir);
+
+	if (single_file) {
+		QFile fCSV(csv_dir.absolutePath() + '/' + sub_dir + '/' + "results.csv");
```
[the remark this answers is not in the document]
Comment (arnost00) [reply libquickevent/libquickeventcore/src/exporters/stageresultscsvexporter.cpp:53]: Ok

File: libquickevent/libquickeventcore/src/exporters/stageresultscsvexporter.cpp (hunk 1 above)
Comment (arnost00) [reply libquickevent/libquickeventcore/src/exporters/stageresultscsvexporter.cpp:55]: Ok. I didn't know it

**Comment by arnost00** [comment]
I added the forgotten option to change the separator for the csv from the UI.

File: libquickevent/libquickeventcore/src/exporters/stageresultscsvexporter.h [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+	StageResultsCsvExporter(QObject *parent = nullptr);
+	void generateCsvMulti();
+	void generateCsvSingle();
+	void setSeparator(QString sep) { m_separator = sep; }
+protected:
+	void exportClasses(bool single_file);
+	void exportClass(int class_id, QTextStream &csv);
+	void exportCsvHeader(QTextStream &csv);
+private:
+	QString m_separator = ";";
```
Comment (fvacek) [inline libquickevent/libquickeventcore/src/exporters/stageresultscsvexporter.h:27]: It was QChar previously, why the change to QString?
